### PR TITLE
Update AWS Calico charts version and add missed endpointslice

### PR DESCRIPTION
**Pull request by haouc**
**What type of PR is this?**
We want to upgrade the aws-calico charts version to 0.3.11 to be synced with eks-charts.

cleanup
**Which issue does this PR fix**:
aws-calico charts version and adding endpointslice into cluster role which was missed in config file.

**What does this PR do / Why do we need it**:

1. We will have a version parity between two repositories.
2. We need add endpointslice 

Tests done:
```
NAMESPACE         NAME                                       READY   STATUS    RESTARTS   AGE
calico-system     calico-kube-controllers-685b68c7c4-xkbds   1/1     Running   0          81s
calico-system     calico-node-42r8c                          1/1     Running   0          81s
calico-system     calico-node-4dv9m                          1/1     Running   0          80s
calico-system     calico-node-bcfvs                          1/1     Running   0          80s
calico-system     calico-node-rffdv                          1/1     Running   0          81s
calico-system     calico-node-t7rm5                          1/1     Running   0          81s
calico-system     calico-typha-6667d47744-5dwxn              1/1     Running   0          73s
calico-system     calico-typha-6667d47744-776jw              1/1     Running   0          73s
calico-system     calico-typha-6667d47744-d699c              1/1     Running   0          81s
kube-system       aws-node-2qblk                             1/1     Running   0          21h
kube-system       aws-node-9qqdk                             1/1     Running   0          21h
kube-system       aws-node-jgbqn                             1/1     Running   0          21h
kube-system       aws-node-kwrqg                             1/1     Running   0          21h
kube-system       aws-node-lhxtk                             1/1     Running   0          21h
kube-system       coredns-85d5b4454c-9mszt                   1/1     Running   0          21h
kube-system       coredns-85d5b4454c-jgwxw                   1/1     Running   0          21h
kube-system       kube-proxy-7rcks                           1/1     Running   0          21h
kube-system       kube-proxy-d7d9s                           1/1     Running   0          21h
kube-system       kube-proxy-jksrr                           1/1     Running   0          21h
kube-system       kube-proxy-llj4n                           1/1     Running   0          21h
kube-system       kube-proxy-mqsp5                           1/1     Running   0          21h
tigera-operator   tigera-operator-6fbb48778f-fss5h           1/1     Running   0          85s
```
```
kubectl get clusterrole tigera-operator -oyaml | grep -e "- discovery.k8s.io" -A5
  - discovery.k8s.io
  resources:
  - endpointslices
  verbs:
  - watch
  - list
```
Star Policy Demo was tested and passed.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

yes
**Automation added to e2e**:

no
**Will this PR introduce any new dependencies?**:

no
**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
no

**Does this change require updates to the CNI daemonset config files to work?**:

no
**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by haouc** [comment]
@tmjd could you review this change. I think the endpointslice was missed from `helm template` when it was added some time before. Thanks.

File: config/master/calico-operator.yaml [hunk 1]
```diff
@@ -4942,6 +4942,14 @@ rules:
       - update
       - delete
       - watch
+  # EndpointSlices are used for Service-based network policy rule
+  # enforcement.
+  - apiGroups: ["discovery.k8s.io"]
```
Comment (tmjd) [inline config/master/calico-operator.yaml:4947]: I don't see endpointslices RBAC in [v3.19](https://github.com/projectcalico/calico/blob/release-v3.19/_includes/charts/tigera-operator/templates/tigera-operator/02-role-tigera-operator.yaml) or in [master](https://github.com/projectcalico/calico/blob/master/calico/_includes/charts/tigera-operator/templates/tigera-operator/02-role-tigera-operator.yaml) so I'm wondering why that would be needed.

I also took a look at the operator code and I don't see any list or watch for endpointslices. The only endpointslices references I found in the operator code is for creating the RBAC for giving permissions to node and typha for endpointslices, and that wouldn't require the operator itself to have endpointslices access.

Comment (haouc) [reply config/master/calico-operator.yaml:4947]: @tmjd this role was added in [this PR](https://github.com/aws/amazon-vpc-cni-k8s/pull/1678). For some reason, the manifest wasn't templated in the PR. Can you take a look if this need to be added into Calico code as well? Thanks.

Comment (tmjd) [reply config/master/calico-operator.yaml:4947]: That PR was based on another PR ([eks-charts#610](https://github.com/aws/eks-charts/pull/610)) that was adding permissions to calico-node. This PR adds permissions to tigera-operator. You can see that the [operator v1.20.1 does give the same endpointslices permission](https://github.com/tigera/operator/blob/v1.20.1/pkg/render/node.go#L252) to the calico-node role, this is the operator version that the [values file includes](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/charts/aws-calico/values.yaml#L4).

I've commented on the PR you linked that it shouldn't have made any difference.

**Pull request merged**